Add EditModuleCommand

wangqinNick opened the pull request.

SimJunYou commented:
Seems like there's a lot of changes made... Can we get a more detailed description?

wangqinNick commented:
> Seems like there's a lot of changes made... Can we get a more detailed description?
For Command part, I only added the editModuleCommand
For ModuleManager part, I only added a function to change the module code of the target module.
For Parser part, I only added a regex to match for editModuleCommand, and a prepareEditModuleCommand() to deal with user inputs
In the end, I added some Junit tests since it is the only way to test. 

PS: I edited your Module class and added a module description attribute
